docs/README.rst

```diff
@@ -0,0 +1,157 @@
+study of cephalopods.
+
+
+Overview
+========
+
+The general mode of operation of ``teuthology`` is to remotely orchestrate
+operations on remote hosts over SSH, as implemented by `Paramiko
+<http://www.lag.net/paramiko/>`__. A typical `job` consists of multiple nested
+`tasks`, each of which perform operations on a remote host over the network.
+
+When testing, it is common to group many `jobs` together to form a `test run`.
+
+If you are new to teuthology and simply want to run existing tests, check out
+:ref:`intro_testers`
+
+
+Provided Utilities
+==================
+* ``teuthology`` - Run individual jobs
+* ``teuthology-coverage`` - Analyze code coverage via lcov
+* ``teuthology-kill`` - Kill running jobs or entire runs
+* ``teuthology-lock`` - Lock, unlock, and update status of machines
+* ``teuthology-ls`` - List job results by examining an archive directory
+* ``teuthology-openstack`` - Use OpenStack backend (wrapper around ``teuthology-suite``)
+* ``teuthology-nuke`` - Attempt to return a machine to a pristine state
+* ``teuthology-queue`` - List, or delete, jobs in the queue
+* ``teuthology-report`` - Submit test results to a web service (we use `paddles <https://github.com/ceph/paddles/>`__)
+* ``teuthology-results`` - Examing a finished run and email results
+* ``teuthology-schedule`` - Schedule a single job
+* ``teuthology-suite`` - Schedule a full run based on a suite (see `suites` in `ceph-qa-suite <https://github.com/ceph/ceph-qa-suite>`__)
+* ``teuthology-updatekeys`` - Update SSH host keys for a mchine
+* ``teuthology-worker`` - Worker daemon to monitor the queue and execute jobs
+
+For a description of the distinct services that utilities interact with see
+:ref:`components`.
+
+Installation
+============
+
+See :ref:`installation_and_setup`.
+
+
+Infrastructure
+==============
+
+The examples in this document are based on the lab machine configuration used
+by the Red Hat Ceph development and quality assurance teams 
+(see :ref:`lab_setup`). Other instances of a Ceph Lab being used in a
+development or testing environment may differ from these examples.
+
+
+Detailed test configuration
+===========================
+
+See :ref:`detailed_test_config`.
+
+
+Virtual Machine Support
+=======================
+
+For OpenStack support, see :ref:`openstack-backend`
+
+For 'vps' support using `downburst <https://github.com/ceph/downburst>`__, see
+:ref:`downburst_vms`
+
+
+Test Suites
+===========
+
+Most of the current teuthology test suite execution scripts automatically
+download their tests from the master branch of the appropriate github
+repository.  People who want to run experimental test suites usually modify the
+download method in the ``teuthology/task`` script to use some other branch or
+repository. This should be generalized in later teuthology releases.
+Teuthology QA suites can be found in ``src/ceph-qa-suite``. Make sure that this
+directory exists in your source tree before running the test suites.
```

Comment:
It is not mandatory to clone the ceph-qa-suite, if not done, teuthology will get the master suite in user's src folder